src/stdlib/SDL_string.c

```diff
@@ -1774,6 +1774,20 @@ static size_t SDL_PrintFloat(char *text, size_t maxlen, SDL_FormatInfo *info, do
     return length;
 }
 
+static size_t SDL_PrintPointer(char *text, size_t maxlen, SDL_FormatInfo *info, const void *value)
+{
+    char num[130];
+    size_t length;
+
+    if (!value) {
+        return SDL_PrintString(text, maxlen, info, NULL);
+    }
+
+    SDL_ulltoa((unsigned long long)(uintptr_t)value, num, 16);
```

Comment:
I wonder whether this works correctly on 128 bit pointers (i.e. Arm's CHERI: https://github.com/libsdl-org/SDL/pull/4552)

Also this particular patch needed in SDL2 too ?

Reply:
I believe unsigned long long will be 128 bits on that architecture.

Yes, this is a good patch for SDL2.

Reply:
Done!